sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/EncryptionAlgorithm.cs

```diff
@@ -2,44 +2,81 @@
+
         /// <summary>
         /// RSA-OAEP
         /// </summary>
-        RSAOAEP,
+        public static readonly EncryptionAlgorithm RsaOaep = new EncryptionAlgorithm(RSAOAEP);
 
         /// <summary>
         /// RSA-15
         /// </summary>
-        RSA15,
+        public static readonly EncryptionAlgorithm Rsa15 = new EncryptionAlgorithm(RSA15);
 
         /// <summary>
         /// RSA-OAEP256
         /// </summary>
-        RSAOAEP256
-    }
+        public static readonly EncryptionAlgorithm RsaOaep256 = new EncryptionAlgorithm(RSAOAEP256);
 
-    internal static class EncryptionAlgorithmExtensions
-    {
-        public static string GetName(this EncryptionAlgorithm algorithm)
-        {
-            switch (algorithm)
-            {
-                case EncryptionAlgorithm.RSAOAEP:
-                    return "RSA-OAEP";
-                case EncryptionAlgorithm.RSA15:
-                    return "RSA1_5";
-                case EncryptionAlgorithm.RSAOAEP256:
-                    return "RSA-OAEP-256";
-                default:
-                    return null;
-            }
-        }
+        /// <summary>
+        /// Determines if two <see cref="EncryptionAlgorithm"/> values are the same.
+        /// </summary>
+        /// <param name="a">The first <see cref="EncryptionAlgorithm"/> to compare.</param>
+        /// <param name="b">The second <see cref="EncryptionAlgorithm"/> to compare.</param>
+        /// <returns>True if <paramref name="a"/> and <paramref name="b"/> are the same; otherwise, false.</returns>
+        public static bool operator ==(EncryptionAlgorithm a, EncryptionAlgorithm b) => a.Equals(b);
+
+        /// <summary>
+        /// Determines if two <see cref="EncryptionAlgorithm"/> values are different.
+        /// </summary>
+        /// <param name="a">The first <see cref="EncryptionAlgorithm"/> to compare.</param>
+        /// <param name="b">The second <see cref="EncryptionAlgorithm"/> to compare.</param>
+        /// <returns>True if <paramref name="a"/> and <paramref name="b"/> are different; otherwise, false.</returns>
+        public static bool operator !=(EncryptionAlgorithm a, EncryptionAlgorithm b) => !a.Equals(b);
+
+        /// <summary>
+        /// Converts a string to a <see cref="EncryptionAlgorithm"/>.
+        /// </summary>
+        /// <param name="value">The string value to convert.</param>
+        public static implicit operator EncryptionAlgorithm(string value) => new EncryptionAlgorithm(value);
+
+        /// <summary>
+        /// Converts a <see cref="EncryptionAlgorithm"/> to a string.
+        /// </summary>
+        /// <param name="value">The <see cref="EncryptionAlgorithm"/> to convert.</param>
+        public static implicit operator string(EncryptionAlgorithm value) => value._value;
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj) => obj is EncryptionAlgorithm other && Equals(other);
```

Comment:
This should have EditorBrowsable false

Reply:
I added this to Equals, GetHashCode and ToString for the added enum like structs.